models/repo_list.go

```diff
@@ -97,21 +97,40 @@ type SearchRepoOptions struct {
 	// Owner in we search search
 	//
 	// in: query
-	OwnerID     int64  `json:"uid"`
-	Searcher    *User  `json:"-"` //ID of the person who's seeking
-	OrderBy     string `json:"-"`
-	Private     bool   `json:"-"` // Include private repositories in results
-	Collaborate bool   `json:"-"` // Include collaborative repositories
-	Starred     bool   `json:"-"`
-	Page        int    `json:"-"`
-	IsProfile   bool   `json:"-"`
+	OwnerID     int64         `json:"uid"`
+	Searcher    *User         `json:"-"` //ID of the person who's seeking
+	OrderBy     SearchOrderBy `json:"-"`
+	Private     bool          `json:"-"` // Include private repositories in results
+	Collaborate bool          `json:"-"` // Include collaborative repositories
+	Starred     bool          `json:"-"`
+	Page        int           `json:"-"`
+	IsProfile   bool          `json:"-"`
 	// Limit of result
 	//
 	// maximum: setting.ExplorePagingNum
 	// in: query
 	PageSize int `json:"limit"` // Can be smaller than or equal to setting.ExplorePagingNum
 }
 
+//SearchOrderBy is used to sort the result
+type SearchOrderBy string
+
+func (s SearchOrderBy) String() string {
```

Comment:
Actually just reviewed it again and does that function is needed? Why not to use string(x) where needed?

Reply:
Because you don't need to care what type SearchOrderBy is (try to think about it like you don't know that it's only string) and this way it implements `Stringer` interface. `string(x)` will not work without other changes if you change SearchOrderBy to int for example. Changing constants should not affect other code.